src/gas/mod.rs

```diff
@@ -222,43 +223,6 @@ impl Counter {
 	}
 }
 
-fn inject_grow_counter(instructions: &mut elements::Instructions, grow_counter_func: u32) -> usize {
-	use parity_wasm::elements::Instruction::*;
-	let mut counter = 0;
-	for instruction in instructions.elements_mut() {
-		if let GrowMemory(_) = *instruction {
-			*instruction = Call(grow_counter_func);
-			counter += 1;
-		}
-	}
-	counter
-}
-
-fn add_grow_counter(module: elements::Module, rules: &rules::Set, gas_func: u32) -> elements::Module {
-	use parity_wasm::elements::Instruction::*;
-
-	let mut b = builder::from_module(module);
-	b.push_function(
-		builder::function()
-			.signature().params().i32().build().with_return_type(Some(elements::ValueType::I32)).build()
-			.body()
-				.with_instructions(elements::Instructions::new(vec![
-					GetLocal(0),
-					GetLocal(0),
-					I32Const(rules.grow_cost() as i32),
-					I32Mul,
-					// todo: there should be strong guarantee that it does not return anything on stack?
-					Call(gas_func),
-					GrowMemory(0),
-					End,
-				]))
-				.build()
-			.build()
-	);
-
-	b.build()
-}
-
```

Comment:
I suggest keeping existing functions as is, and add a parallel infra for global based gas counting. It's important that we can run both versions side by side and then compare them aginst each other. 

More generally, when doing any change, it's almost always faster to *first* start adding new code alongside the old one, *then* have both versions working together, *then* migrate users from the old version to the new version, *then* finally remove the old code. There's a lot of indirect benefit in having working code during every step of the change.

Reply:
Got it! That's good to know